apps/designer/app/designer/features/topbar/sync-status.tsx

```diff
@@ -1,50 +1,62 @@
 import { useStore } from "@nanostores/react";
+import { atom } from "nanostores";
 import {
-  theme,
   Flex,
-  keyframes,
   AccessibleIcon,
-  styled,
-  Box,
+  rawTheme,
+  Tooltip,
 } from "@webstudio-is/design-system";
-import { CheckIcon, DotsHorizontalIcon } from "@webstudio-is/icons";
-import { syncStatus } from "~/designer/shared/sync-server";
+import { CloudIcon } from "@webstudio-is/icons";
+import { useEffect } from "react";
+import { syncStatus } from "~/designer/shared/sync";
 
-const iconSize = 15;
+const isOnlineStore = atom(false);
 
-const StyledCheckIcon = styled(CheckIcon, {
-  width: iconSize,
-  height: iconSize,
-  background: theme.colors.green9,
-  borderRadius: theme.borderRadius.round,
-});
-
-const ellipsisKeyframes = keyframes({
-  to: { width: iconSize },
-});
-
-const AnimatedDotsIcon = () => {
-  return (
-    <Flex direction="column" css={{ width: iconSize }}>
-      <Box
-        css={{
-          animation: `${ellipsisKeyframes} steps(4,end) 900ms infinite`,
-          width: 0,
-          overflow: "hidden",
-        }}
-      >
-        <DotsHorizontalIcon width="12" height="12" />
-      </Box>
-    </Flex>
-  );
+const useSetOnline = () => {
```

Comment:
Why use if it's not hook? atom use useSyncExternalStore underhood too. Why not use it?

Reply:
Won't need to have useEffect

Reply:
I don't want to use useSyncExternalStore in the application logic, it is too low level

Reply:
changing the name

Reply:
Isn't useEffect low level too?

Reply:
it is, but at least everyone is intimately familiar with its interface

Reply:
I don't feel comfortable using syncExternalStore directly

Reply:
They both do exactly same: synchronization. useSyncExternalStore syncrhonize external store, useEffect synchronize dom.

Reply:
yeah, though external store also runs serverside, but I only need this particular thing to run in frontend only

Reply:
I just don't feel the need for it here

Reply:
Don't see the benefit

Reply:
over atom we got the store anyways, so unless you want to fix some problem here, lets not waste time